### PR TITLE
AKU-410: Updates required for dashlet prototype

### PR DESCRIPTION
**Pull request by draperd**
This PR addresses https://issues.alfresco.com/jira/browse/AKU-410 in providing the necessary updates to support the dashlet code attached to that JIRA ticket.

### CODE REVIEW

File: aikau/src/main/resources/alfresco/documentlibrary/_AlfDndDocumentUploadMixin.js [hunk 1]
```diff
@@ -547,7 +546,7 @@ define(["dojo/_base/declare",
             try
             {
                config = {
-                  destination: this.currentItem.nodeRef
+                  destination: this.currentItem.node.nodeRef
```
Comment (MartinDoyleUK) [inline aikau/src/main/resources/alfresco/documentlibrary/_AlfDndDocumentUploadMixin.js]: node is definitely there?

File: aikau/src/main/resources/alfresco/documentlibrary/_AlfDndDocumentUploadMixin.js [hunk 2]
```diff
@@ -563,7 +562,7 @@ define(["dojo/_base/declare",
             try
             {
                config = {
-                  updateNodeRef: this.currentItem.nodeRef,
+                  updateNodeRef: this.currentItem.node.nodeRef,
```
Comment (MartinDoyleUK) [inline aikau/src/main/resources/alfresco/documentlibrary/_AlfDndDocumentUploadMixin.js]: again, node is definitely there?

File: aikau/src/main/resources/alfresco/renderers/_ItemLinkMixin.js [hunk 3]
```diff
@@ -114,7 +113,7 @@ define(["dojo/_base/declare",
                      // TODO: It'll be necessary to get the actual actionUrls - but currently it's to tangled to untangle easily
                      //var actionUrls = this.getActionUrls(item);
                      var actionUrls = this.getActionUrls(item);
-                     if (jsNode.isLink && jsNode.linkedNode.isContainer)
+                     if (item.node.isLink && item.nod.linkedNode.isContainer)
```
Comment (MartinDoyleUK) [inline aikau/src/main/resources/alfresco/renderers/_ItemLinkMixin.js]: `item.nod`?

Comment (draperd) [inline aikau/src/main/resources/alfresco/renderers/_ItemLinkMixin.js]: Good catch

File: aikau/src/main/resources/alfresco/documentlibrary/_AlfDndDocumentUploadMixin.js (hunk 1 above)
Comment (draperd) [inline aikau/src/main/resources/alfresco/documentlibrary/_AlfDndDocumentUploadMixin.js]: That's why it's in a try/catch... it's *should* be :)

File: aikau/src/main/resources/alfresco/documentlibrary/_AlfDndDocumentUploadMixin.js (hunk 2 above)
Comment (draperd) [inline aikau/src/main/resources/alfresco/documentlibrary/_AlfDndDocumentUploadMixin.js]: As with previous, it *should* be and is protected with try/catch

**Pull request merged**